File: src/SimpleLevelEditor.Formats.Types/DataFormatExtensions.fs
[<AutoOpen>]
module SimpleLevelEditor.Formats.Types.DataFormatExtensions

open System
open System.Globalization
open System.Numerics

type Boolean with
    member this.ToDisplayString =
        this.ToString(CultureInfo.InvariantCulture).ToLowerInvariant()

type Byte with
    member this.ToDisplayString =
        this.ToString(CultureInfo.InvariantCulture)

type Int32 with
    member this.ToDisplayString =
        this.ToString(CultureInfo.InvariantCulture)

type Single with
    member this.ToDisplayString =
        this.ToString(CultureInfo.InvariantCulture)

type Vector2 with
    member this.ToDisplayString =
        $"%s{this.X.ToDisplayString} %s{this.Y.ToDisplayString}"

type Vector3 with
    member this.ToDisplayString =
        $"%s{this.X.ToDisplayString} %s{this.Y.ToDisplayString} %s{this.Z.ToDisplayString}"

type Vector4 with
    member this.ToDisplayString =
        $"%s{this.X.ToDisplayString} %s{this.Y.ToDisplayString} %s{this.Z.ToDisplayString} %s{this.W.ToDisplayString}"

File: src/SimpleLevelEditor.Formats.Core/DataFormatExtensions.cs
using System.Globalization;
using System.Numerics;

namespace SimpleLevelEditor.Formats.Core;

public static class DataFormatExtensions
{
	public static string ToDisplayString(this bool value)
	{
		return value ? "true" : "false";
	}

	public static string ToDisplayString(this byte value)
	{
		return value.ToString(CultureInfo.InvariantCulture);
	}

	public static string ToDisplayString(this int value)
	{
		return value.ToString(CultureInfo.InvariantCulture);
	}

	public static string ToDisplayString(this float value)
	{
		return value.ToString(CultureInfo.InvariantCulture);
	}

	public static string ToDisplayString(this Vector2 value)
	{
		return $"{value.X.ToDisplayString()} {value.Y.ToDisplayString()}";
	}

	public static string ToDisplayString(this Vector3 value)
	{
		return $"{value.X.ToDisplayString()} {value.Y.ToDisplayString()} {value.Z.ToDisplayString()}";
	}

	public static string ToDisplayString(this Vector4 value)
	{
		return $"{value.X.ToDisplayString()} {value.Y.ToDisplayString()} {value.Z.ToDisplayString()} {value.W.ToDisplayString()}";
	}
}
